multi_physics/QED/include/picsar_qed/qed_commons.h

```diff
@@ -0,0 +1,144 @@
+#ifndef PICSAR_MULTIPHYSICS_QED_COMMONS
+#define PICSAR_MULTIPHYSICS_QED_COMMONS
+
+/**
+ * This header file should be included by all the source files of the library,
+ * since it containes some common definitions.
+ */
+
+
+/**
+ * The core functions of the library run also on GPUs.
+ * In order to do so, the user has to define PXRMP_GPU as follows:
+ * #define PXRMP_WITH_GPU
+ * Otherwise the code is compiled without GPU support
+ */
+//#define PXRMP_WITH_GPU
+
+
+/**
+* If the code is compiled with GPU support,
+ * PICSAR sets PXRMP_GPU_QUALIFIER to `__host__ __device__`
+ * However, the user can override this by defining PXRMP_GPU_QUALIFIER
+ * manually.
+ */
+#ifdef PXRMP_WITH_GPU
+    #ifndef PXRMP_GPU_QUALIFIER
+        #define PXRMP_GPU_QUALIFIER __host__ __device__
+    #endif
+#else
+    #define PXRMP_GPU_QUALIFIER
+#endif
+
+
+/**
+ * If GPU support is not enabled, there should be no need to use
+ * GPU friendly PICSAR arrays. However, the user can force their
+ * use with:
+ * #define PXRMP_FORCE_PICSAR_ARRAY
+ * This should be done only for debug purposes.
+ */
+#if defined(PXRMP_WITH_GPU) || defined(PXRMP_FORCE_PICSAR_ARRAY)
+    #define PXRMP_ENABLE_GPU_FRIENDLY_ARRAY
+#endif
+
+
+/**
+ * The user can explicitly define a restrict keyword by doing
+ * #define PXRMP_RESTRICT __restrict__
+ * otherwise a choice based on the operating system is made.
+ */
+#ifndef PXRMP_RESTRICT
+  #ifdef _WIN32
+    #define PXRMR_RESTRICT __restrict
+  #else
+    #define PXRMR_RESTRICT __restrict__
+  #endif
+#endif
+
+
+/**
+ * The user can explicitly define a force inline keyword by doing
+ * #define PXRMP_FORCE_INLINE __forceinline__
+ * otherwise a choice based on the compiler is made
+ */
+#ifndef PXRMP_FORCE_INLINE
+  #if defined(__CUDA_ARCH__)
+    #define PXRMP_FORCE_INLINE __forceinline__
+  #elif defined(__INTEL_COMPILER)
+    #define PXRMP_FORCE_INLINE inline __attribute__((always_inline))
+  #elif defined(__clang__)
+    #define PXRMP_FORCE_INLINE inline __attribute__((always_inline))
+  #elif defined(__GNUC__)
+    #define PXRMP_FORCE_INLINE inline __attribute__((always_inline))
+  #elif defined(__ibmxl__)
+    #define PXRMP_FORCE_INLINE inline __attribute__((always_inline))
+  #else
+    #define PXRMP_FORCE_INLINE inline
+  #endif
+#endif
+
+
+/**
+ * By doing
+ * #define PXRMP_USE_CXX17_FOR_SPECIAL_FUNCTIONS
+ * the user can choose to use special functions provided by the C++17
+ * standard library if C++17 is available. Otherwise, special functions
+ * provided by Boost are used.
+ */
+ #ifdef PXRMP_USE_CXX17_FOR_SPECIAL_FUNCTIONS
+     #if !(__cplusplus > 201402L)
```

Comment:
You mean
```suggestion
     #if __cplusplus < 201703L
```
https://gist.github.com/ax3l/53db9fa8a4f4c21ecc5c4100c0d93c94#meaning-of-__cplusplus

Reply:
yeah, you solution is better